include/zephyr/devicetree.h

```diff
@@ -911,6 +913,95 @@
 	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop), \
 		(DT_STRING_UPPER_TOKEN(node_id, prop)), (default_value))
 
+/**
+ * @brief Get a string-array item value as a token.
+ *
+ * This removes "the quotes" from string-valued item, and converts
+ * non-alphanumeric characters to underscores. That can be useful, for example,
+ * when programmatically using the value to form a C variable or code.
+ *
+ * DT_STRING_TOKEN_BY_IDX() can only be used for properties with
+ * string-array type.
+ *
+ * It is an error to use DT_STRING_TOKEN_BY_IDX() in other circumstances.
+ *
+ * Example devicetree fragment:
+ *
+ *     n1: node-1 {
+ *             prop = "f1", "F2";
+ *     };
+ *     n2: node-2 {
+ *             prop = "123 foo", "456 FOO";
+ *     };
+ *
+ * Example bindings fragment:
+ *
+ *     properties:
+ *       prop:
+ *         type: string-array
+ *
+ * Example usage:
+ *
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n1), prop, 0) // f1
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n1), prop, 1) // F2
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n2), prop, 0) // 123_foo
+ *     DT_STRING_TOKEN_BY_IDX(DT_NODELABEL(n2), prop, 1) // 456_FOO
+ *
+ * For more notices see @ref DT_STRING_TOKEN.
```

Comment:
```suggestion
 * For more information, see @ref DT_STRING_TOKEN.
```

Reply:
Unresolving this one since there are other cases in this PR where the same change needs to be done. Sorry I was vague. This was meant as an example of what to do in general, so all the cases where "notices" was used before, "information," should be used instead.